### ISSUE TITLE
TravisCI did not run recently

### ISSUE THREAD

**Issue by eregon** [kwilczynski/ruby-magic#15]
See https://travis-ci.com/github/kwilczynski/ruby-magic/builds

https://travis-ci.com/github/kwilczynski/ruby-magic/requests shows:
> Owner kwilczynski does not have enough credits.

So I guess TravisCI is really not supporting OSS anymore with such limited build time (I see the same for my TravisCI account).

I think most people migrate to GitHub Actions because it's unrestricted, faster, more reliable and better integrated.

**Comment by kwilczynski** [comment]
Hi @eregon,

Indeed! Sincere apologies :( I sadly can't afford to pay for the commercial plan :( $70 per month is unfortunately outside of my financial reach.

I wish Travis CI continued the offer support for Open Source (and public) projects. I mailed their support raising a question about this, but sadly haven't got any response as of yet - don't expect any, actually. Well, such is their prerogative not to offer free compute time.

I discussed this recently with @stanhu too and he recommended GitLab CI. :)

Krzysztof

**Comment by eregon** [comment]
> I discussed this recently with @stanhu too and he recommended GitLab CI. :)

I don't know much about GitLab CI, but since this project is on GitHub I'd expect GitHub Actions to be better integrated.
Of course I'm biased since I maintain https://github.com/ruby/setup-ruby :)

**Comment by kwilczynski** [comment]
Hello everyone,

A small update.

Despite reaching out to Travis CI many times via the `support@travis-ci.com` and on Twitter, they never responded, so we are going to be moving away from them completely.

It's a sad day, as Travis CI used to be to go-to platform for Open Source projects.

Krzysztof

**Comment by eregon** [comment]
Yes, this seems to happen to many people still using TravisCI recently it seems:
https://twitter.com/smarr/status/1380195827591876612
https://twitter.com/sh0dan/status/1380555125714997253

Maybe they reduced open-source credits or so significantly? (and before that it was very long queue time which means quite a few projects migrated already).

**Comment by kwilczynski** [comment]
Hi @eregon,

> Yes, this seems to happen to many people still using TravisCI recently it seems:
> https://twitter.com/smarr/status/1380195827591876612
> https://twitter.com/sh0dan/status/1380555125714997253

Oh I see! Not good...

> Maybe they reduced open-source credits or so significantly? (and before that it was very long queue time which means quite a few projects migrated already).

I am not sure. I never heard back from them. I assume Open Source is no longer their (or the new owner, rather) concern.

Krzysztof

**Comment by stanhu** [comment]
GitLab Ci seems to be working well: https://gitlab.com/stanhu/ruby-magic/-/pipelines/284133549 😄

**Comment by stanhu** [comment]
The only catch is that GitLab CI uses a Webhook to trigger new commits, but since I don't have access here to install a WebHook the CI runs only as often as the pull mirror updates.

**Comment by kwilczynski** [comment]
Hi @stanhu, sorry for the delay!

I am happy to move to GitLab CI, and set whatever webhook would be needed. No problem! Let me know what would need to be done and I will see it done.

Interestingly, since I have personal account, you can't manage my repository (sorry about that), but if this would be an organisation or a paid account, then you could be added with enough permission to change anything you see fit.

At the moment I don't have the budget to do it (since the price is a lot higher), but I am going to think about it. To be honest, perhaps I should move to GitLab with everything.

Krzysztof

**Comment by kwilczynski** [comment]
Hello everyone,

A update on Travis CI.

I've noticed that Travis CI added/replenished some of the free credits eventually, albeit they are about to run out shortly anyway. I find out when I got an e-mail about failed job from Travis CI.

We are definitely still on the lookout for new CI place, and @stanhu probably found us one on GitLab.

Krzysztof

**Comment by kwilczynski** [comment]
Hello,

I wanted to add, that @erikh also offered help setting up and using [tinyCI](https://tinyci.org/), to remain vendor agnostic.

Krzysztof

**Comment by flavorjones** [comment]
FWIW I'm in the process of moving some of my other projects to Github Actions, my experience with it has been good so far.

**Comment by kwilczynski** [comment]
Hi @flavorjones,

> FWIW I'm in the process of moving some of my other projects to Github Actions, my experience with it has been good so far.

I saw it! I am following your work on Nokogiri and GitHub actions closely, and I also think this would work for us, especially since this project is also so much simpler.

Krzysztof
